src/tests/Interop/MarshalAPI/FunctionPointer/FunctionPointer.cs

```diff
@@ -111,4 +112,53 @@ public static void RunGetDelForOutIntTest()
         }
         Assert.Equal(expectedValue, outVar);
     }
+
+    [UnmanagedCallersOnly(CallConvs = [typeof(CallConvCdecl)])]
+    static int UnmanagedExportedFunction(float arg)
+    {
+        return Convert.ToInt32(arg);
+    }
+
+    class GenericCaller<T>
+    {
+        internal static unsafe T GenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged<U, T>)fnptr)(arg);
+        }
+    }
+
+    [Fact]
+    public static void RunGenericFunctionPointerTest()
+    {
+        Console.WriteLine($"Running {nameof(RunGenericFunctionPointerTest)}...");
+        int outVar = 0;
+        int expectedValue = 42;
+        unsafe
+        {
+            outVar = GenericCaller<int>.GenericCalli((delegate* unmanaged[Cdecl]<float, int>)&UnmanagedExportedFunction, 42.0f);
+        }
+        Assert.Equal(expectedValue, outVar);
+    }
+
```

Comment:
Also, add tests where the arguments itself are generic. For example:
```
foo<string>(null);

static void foo<T>(void* p)
{
    var pfn = (delegate *unmanaged<BlittableGeneric<T>>)(p);
    pfn();
}

}

struct BlittableGeneric<T>
{
   int x;
}
```
I think this case is not handled correctly in the current state of the PR.